config/memory_statistics.py

```diff
@@ -0,0 +1,114 @@
+
+
+def get_memory_statistics_table(db):
+    """Get the MEMORY_STATISTICS table from the database."""
+    return db.get_table("MEMORY_STATISTICS")
+
+
+@memory_statistics.command(name="enable", short_help="Enable the Memory Statistics feature")
+def memory_statistics_enable():
+    """Enable the Memory Statistics feature"""
+    db = ConfigDBConnector()
+    db.connect()
+
+    memory_statistics_table = get_memory_statistics_table(db)
+    if not check_memory_statistics_table_existence(memory_statistics_table):
+        return  # Exit gracefully on error
+
+    try:
+        db.mod_entry("MEMORY_STATISTICS", "memory_statistics", {"enabled": "true", "disabled": "false"})
+        click.echo("Memory Statistics feature enabled.")
+    except Exception as e:
+        click.echo(f"Error enabling Memory Statistics feature: {str(e)}", err=True)
+
+    click.echo("Save SONiC configuration using 'config save' to persist the changes.")
+
+
+@memory_statistics.command(name="disable", short_help="Disable the Memory Statistics feature")
+def memory_statistics_disable():
+    """Disable the Memory Statistics feature"""
+    db = ConfigDBConnector()
+    db.connect()
+
+    memory_statistics_table = get_memory_statistics_table(db)
+    if not check_memory_statistics_table_existence(memory_statistics_table):
+        return  # Exit gracefully on error
+
+    try:
+        db.mod_entry("MEMORY_STATISTICS", "memory_statistics", {"enabled": "false", "disabled": "true"})
+        click.echo("Memory Statistics feature disabled.")
+    except Exception as e:
+        click.echo(f"Error disabling Memory Statistics feature: {str(e)}", err=True)
+
+    click.echo("Save SONiC configuration using 'config save' to persist the changes.")
+
+
+@memory_statistics.command(name="retention-period", short_help="Configure the retention period for Memory Statistics")
+@click.argument('retention_period', metavar='<retention_period>', required=True, type=int)
+def memory_statistics_retention_period(retention_period):
+    """Set the retention period for Memory Statistics"""
+    db = ConfigDBConnector()
+    db.connect()
+
+    memory_statistics_table = get_memory_statistics_table(db)
+    if not check_memory_statistics_table_existence(memory_statistics_table):
+        return  # Exit gracefully on error
+
+    try:
+        db.mod_entry("MEMORY_STATISTICS", "memory_statistics", {"retention_period": retention_period})
+        click.echo(f"Memory Statistics retention period set to {retention_period} days.")
+    except Exception as e:
+        click.echo(f"Error setting retention period: {str(e)}", err=True)
+
+    click.echo("Save SONiC configuration using 'config save' to persist the changes.")
+
+
+@memory_statistics.command(name="sampling-interval", short_help="Configure the sampling interval for Memory Statistics")
+@click.argument('sampling_interval', metavar='<sampling_interval>', required=True, type=int)
+def memory_statistics_sampling_interval(sampling_interval):
```

Comment:
>sampling_interval [](http://example.com/codeflow?start=40&length=17)

Please add verify and validation for input parameter.

Reply:
I’ve added input validation for the sampling_interval parameter. The value is now checked to ensure it falls between 3 and 15 minutes, with clear feedback provided for any invalid input.